numpy/core/fromnumeric.py

```diff
@@ -445,6 +445,10 @@ def put(a, ind, v, mode='raise'):
     array([ 0,  1,  2,  3, -5])
 
     """
+    if not isinstance(a, np.ndarray):
```

Comment:
IMHO using ```_wrapit``` does not help here because I want to enforce that an ```ndarray``` must be passed in.  Because ```np.put``` modifies the ```ndarray``` <b>in-place</b>, passing in a non-```ndarray``` object will do nothing (the array_like object itself is not changed because it has to be cast to an ```ndarray``` object) and will not raise any errors, which is what I am trying to avoid.  This was the issue that I saw with ```np.place```.